starlite/datastructures/headers.py

```diff
@@ -0,0 +1,118 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, ClassVar, Optional
+
+from pydantic import BaseModel, Extra, ValidationError
+
+from starlite.exceptions import ImproperlyConfiguredException
+
+if TYPE_CHECKING:
+    from typing import Any, Dict
+
+
+class Header(BaseModel, ABC):
+    """An abstract type for HTTP headers."""
+
```

Comment:
The `documentation_only` field is missing. You mention it in the documentation, so it should be added here as well.

Reply:
I don't think `documentation_only` is needed in these "extra" headers. Do you see a use case where someone needs to add them only for documentation (OpenAPI) purposes?

Where do I mention it in the documentation?

Reply:
This is meant for the cases where users set headers dynamically. For example, when returning a response instance they create.

Reply:
I'm not sure I understand... 

When users want to use `CacheControlHeader` they'll likely do something like this:

```python
@get(path="/handler", cache_control=CacheControlHeader(max_age=1000))
def my_handler(self) -> None:
    pass
```

Behind the scenes, [we create a `ResponseHeader` instance](https://github.com/seladb/starlite/blob/da0ef9943b2521564772aa7bc4d41ca8e8b90051/starlite/handlers/http.py#L433) with a default value of `documentation_only=False` and add it to the list of `resolved_response_headers`.

In this scenario, what's the use case for creating a `CacheControlHeader(..., documentation_only=True)`?

In other words, why would users want to create a `CacheControlHeader` header if they don't want to return it in the response?

Reply:
They might want to return it in the response, but the view could be dependent on the response itself, so they couldn’t be set at the time when they declare the handler.

In this case, having `documentation_only=True` means they can set that when declaring the handler, but set the "real header" dynamically within the function body.

> Where do I mention it in the documentation?

Disregard that please. You didn’t actually mention it in the docs, I just mixed up some lines there while reading.

Reply:
Sure, added here: https://github.com/starlite-api/starlite/pull/601/commits/2089e9002542eee8d171517424d80bd77999a1fe